chcemvediet/apps/inforequests/tests/test_views/test_mine.py

```diff
@@ -0,0 +1,155 @@
+        successful_inforequests2 = [self._create_successful_inforequest(applicant=self.user2) for i in range(3)]
+        unsuccessful_inforequests2 = [self._create_unsuccessful_inforequest(applicant=self.user2) for i in range(3)]
+
+        self._login_user(self.user1)
+        response = self.client.get(reverse(u'inforequests:mine'))
+        self.assertEqual(response.status_code, 200)
+        self.assertItemsEqual(response.context[u'pending_inforequests'], [])
+        self.assertItemsEqual(response.context[u'drafts'], [])
+        self.assertItemsEqual(response.context[u'successful_inforequests'], [])
+        self.assertItemsEqual(response.context[u'unsuccessful_inforequests'], [])
+
+    def test_related_models_are_prefetched_before_render(self):
+        drafts1 = [self._create_inforequest_draft(applicant=self.user1) for i in range(5)]
+        pending_inforequests1 = [self._create_pending_inforequest(applicant=self.user1) for i in range(4)]
+        successful_inforequests1 = [self._create_successful_inforequest(applicant=self.user1) for i in range(3)]
+        unsuccessful_inforequests1 = [self._create_unsuccessful_inforequest(applicant=self.user1) for i in range(3)]
+
+        # Force view querysets to evaluate before calling render
+        def pre_mock_render(request, template, context):
+            list(context[u'pending_inforequests'])
+            list(context[u'drafts'])
+            list(context[u'successful_inforequests'])
+            list(context[u'unsuccessful_inforequests'])
+
+        self._login_user(self.user1)
+        patterns_single_column = [
+            u'FROM "accounts_profile" WHERE "accounts_profile"."user_id" = %s LIMIT 21',
+            squeeze(u"""
+                SELECT COUNT\(\*\) FROM "mail_message"
+                INNER JOIN "inforequests_inforequestemail" ON \( "mail_message"."id" = "inforequests_inforequestemail"."email_id" \)
+                INNER JOIN "inforequests_inforequest" ON \( "inforequests_inforequestemail"."inforequest_id" = "inforequests_inforequest"."id" \)
+                WHERE \("inforequests_inforequestemail"."type" = %s
+                    AND "inforequests_inforequest"."applicant_id" = %s
+                    AND "inforequests_inforequest"."closed" = %s\)
+                """),
+            u'FROM "invitations_invitationsupply" WHERE "invitations_invitationsupply"."user_id" = %s LIMIT 21',
+        ]
+        # TODO: Optimize N+1 queries in #419
+        patterns_pending_inforequests = [
+            squeeze(u"""
+                FROM "inforequests_action"
+                INNER JOIN "inforequests_branch" ON \( "inforequests_action"."branch_id" = "inforequests_branch"."id" \)
+                WHERE "inforequests_branch"."inforequest_id" = %s
+                ORDER BY "inforequests_action"."created" DESC, "inforequests_action"."id" DESC
+                LIMIT 1
+                """) for _ in range(4)
+        ]
```

Comment:
Preco tu mas `for _ in range(4)` a na osatnych miestach `] * 3`? Nebolo by jednoduchsie aj tu iba `] * 4`?

Reply:
:ok: